module/gui/PackageDock.py

```diff
@@ -40,51 +59,214 @@ def slotDone(self):
-        for url in result:
-            if "\n" or "\t" or "\r" or "\"" or "<" or "'" in url:
-                url = url[:-1]
-            self.widget.box.append("%s " % url)
+        event.ignore()
+    
+    def paintEvent(self, event):
+        QDockWidget.paintEvent(self, event)
+        if self.isFloating():
+            self.geo = self.geometry()
+        self.paintEventCounter += 1
+        if self.paintEventSignal:
+            self.paintEventSignal = False
+            self.emit(SIGNAL("newPackDockPaintEvent"))
+    
+    def moveEvent(self, event):
+        if self.isFloating():
+            self.geo = self.geometry()
 
 class NewPackageWindow(QWidget):
     def __init__(self, dock):
         QWidget.__init__(self)
+        self.log = logging.getLogger("guilog")
         self.dock = dock
-        self.setLayout(QGridLayout())
+        
+        self.setLayout(QVBoxLayout())
         layout = self.layout()
         
         nameLabel = QLabel(_("Name"))
         nameInput = QLineEdit()
         passwordLabel = QLabel(_("Password"))
         passwordInput = QLineEdit()
         
-        linksLabel = QLabel(_("Links in this Package"))
+        linksLabel = QLabel(_("Drop or Paste URLs here") + ":")
         
-        self.box = QTextEdit()
+        self.box = PlainTextEdit()
         self.nameInput = nameInput
+        self.passwordLabel = passwordLabel
         self.passwordInput = passwordInput
         
-        save = QPushButton(_("Create"))
-        parseUri = QPushButton(_("Filter URLs"))
+        self.append = QCheckBox(_("Append URLs"))
+        whatsThis = (self.append.text(), _("Append pasted/dropped text instead of inserting it at the cursor postion."))
+        self.append.setWhatsThis(whatsThisFormat(*whatsThis))
+        whatsThis = (linksLabel.text(), _("One URL (Link) per line.<br>For manual text editing you may want to deactivate") + " '" + self.append.text() + "'.")
+        self.box.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.destQueue     = QRadioButton(_("Queue"))
+        self.destCollector = QRadioButton(_("Collector"))
+        destBtnLayout = QHBoxLayout()
+        destBtnLayout.addWidget(self.destQueue)
+        destBtnLayout.addWidget(self.destCollector)
+        destBtnLayout.addStretch(1)
+        self.destAutoSelect = QCheckBox(_("Select with tab"))
+        
+        self.clear = QPushButton(_("Clear"))
+        whatsThis = (self.clear.text(), _("Clears the URL box.<br>A subsequent click also clears Name and Password."))
+        self.clear.setWhatsThis(whatsThisFormat(*whatsThis))
+        self.filter = QPushButton()
+        
+        self.save = QPushButton(_("Create"))
+        self.save.setIcon(self.dock.appIconSet["add_small"])
+        self.msg = QLabel("ERROR")
+        s = "QWidget { color: crimson; background-color: %s }" % QColor(Qt.gray).name()     # red text color on gray background
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['QColor' may be undefined, or defined from star imports: PyQt4.QtCore, PyQt4.QtGui](https://app.codacy.com/manual/pyLoad/pyload/pullRequest?prid=567903)